WooCommerce/Classes/ViewRelated/Dashboard/JetpackConnectionPackageSites/JetpackBenefitsView.swift

```diff
@@ -65,9 +72,19 @@ struct JetpackBenefitsView: View {
             // Actions
             VStack(spacing: Layout.spacingBetweenCTAs) {
                 // Primary Button to install Jetpack
-                Button(Localization.installAction, action: installAction)
-                    .buttonStyle(PrimaryButtonStyle())
-                    .fixedSize(horizontal: false, vertical: true)
+                Button(Localization.installAction) {
+                    guard !isJetpackCPSite else {
+                        return installAction(nil)
```

Comment:
Nit: Instead of sending back `nil` as the Result, can we add a new Error and send that instead? 

IMO, with a custom error we can add clarity about why `JetpackUser` cannot be returned. Also, by handling the custom error [here](https://github.com/woocommerce/woocommerce-ios/pull/8944/files#diff-b3948b1c713bef31b9d1597d8ff7e6c924e7f843c03939905e5d9bba06d323deR32-R36) we can add clarity about JCP vs Jetpack. 

What do you think?

Reply:
Also done in a320f4daf21f70e8e086023545a31253dd5ddc30.